### DISCUSSION TITLE
Contributing Translations to the Project

### DISCUSSION THREAD

**Discussion by MDeLuise** [MDeLuise/plant-it#148] [Show and tell]
Hi Plant-it community!
In this discussion, we'll explore how you can contribute to the project by adding translations using Flutter's internationalization features, specifically the Application Resource Bundle.

## How to Contribute

Let's walk through an example of adding a new language translation, such as Spanish, to the project:

1. **Identify Language Code**: First, determine the [language code](https://api.flutter.dev/flutter/flutter_localizations/GlobalMaterialLocalizations-class.html) for the desired translation. For Spanish, the language code is `es`.

2. **Create Translation File**: Within the `lib/l10n` folder of the project, create a new file named `app_es.arb`.

3. **Copy and Modify Content**: Copy the content of the existing translation file `app_en.arb` into the newly created `app_es.arb` file. Then, modify the values of the translations as needed to provide accurate translations for the Spanish language.

## Resources and Documentation

For further guidance and information, refer to the official documentation on Flutter's internationalization features and the Application Resource Bundle specification:

- [Flutter Internationalization Features](https://docs.flutter.dev/ui/accessibility-and-internationalization/internationalization)
- [Application Resource Bundle Specification](https://github.com/google/app-resource-bundle/wiki/ApplicationResourceBundleSpecification)

Contributing translations not only helps make the project accessible to a wider audience but also enriches the user experience for users around the world. Thank you for your contributions!

**Comment by ghost** [comment]
Hello,

I suggest to host the translation project on Hosted Weblate. They offer free plans for FOSS software.

https://hosted.weblate.org/
